sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -860,6 +871,92 @@ async def update_tag_properties(
             repository=repository
         )
 
+    @distributed_trace_async
+    async def set_manifest(
+        self,
+        repository: str,
+        manifest: Union[JSON, IO[bytes]],
+        *,
+        tag: Optional[str] = None,
+        media_type: str = OCI_IMAGE_MANIFEST,
+        **kwargs
+    ) -> str:
+        """Set a manifest for an artifact.
+
+        :param str repository: Name of the repository
+        :param manifest: The manifest to set. It can be a JSON formatted dict or seekable stream.
+        :type manifest: dict or IO
+        :keyword tag: Tag of the manifest.
+        :paramtype tag: str or None
+        :keyword media_type: The media type of the manifest. If not specified, this value will be set to
+            a default value of "application/vnd.oci.image.manifest.v1+json". Note: the current known media types are:
+            "application/vnd.oci.image.manifest.v1+json", and "application/vnd.docker.distribution.manifest.v2+json".
+        :paramtype media_type: str
+        :returns: The digest of the set manifest, calculated by the registry.
+        :rtype: str
+        :raises ValueError: If the parameter repository or manifest is None,
+            or the digest in the response does not match the digest of the set manifest.
+        """
+        try:
+            if isinstance(manifest, MutableMapping):
+                data = _UnclosableBytesIO(json.dumps(manifest).encode())
+            else:
+                data = _UnclosableBytesIO(manifest.read())
+            tag_or_digest = tag
+            if tag_or_digest is None:
+                tag_or_digest = _compute_digest(data)
+
+            response_headers = await self._client.container_registry.create_manifest(
+                name=repository,
+                reference=tag_or_digest,
+                payload=data,
+                content_type=media_type,
+                cls=_return_response_headers,
+                **kwargs
+            )
+            digest = response_headers['Docker-Content-Digest']
+            if not _validate_digest(data, digest):
+                raise ValueError("The digest in the response does not match the digest of the set manifest.")
```

Comment:
This doesn't match the sync error message?

Reply:
Updated.